az-digital/az_quickstart
# Closes #2091, #1605, #1772 Add an optional thumbnail image field to AZ News, AZ Event

(Description last updated Apr 24, 2023)

## Description

Closes #2091, #1605, and #1772 with the following changes:

1. Added a Thumbnail Image field and a Use Featured Image as Thumbnail field to the AZ News and AZ Event content types.
2. Added pre-save hooks so that the Thumbnail Image field is automatically set to match the featured image for the node if the checkbox for the Use Featured Image as Thumbnail field is checked.
3. Added a new Grid View with Images view display for AZ Event, as an alternative to the Grid View.
4. Added a new Card Alternate view mode and a new Card Alternate display for AZ Event to support the new view.
5. Updated the Row display for AZ Event to include the thumbnail image displayed with the Square (220X220px) view mode.
6. Added a new Square Aspect Ratio crop type and updated the Square (220x220) image style to allow manual cropping.
8. Updated all AZ News displays using the Card Image or Thumbnail (100 x 100) view modes to use the new Thumbnail Image field instead of the Photo field for that image.
9. Added new content for news and events to the demo module which demonstrates usage of the optional thumbnail field.
10. Added a new process plugin in az_core to allow migrating multiple images to the Event Photos field.
11. Removed unused field groups from the Card display for AZ Event, the Medium Media List display for AZ News, and the Minimal Media List display for AZ News.

## Related issues




#2091, #1605, #1772

## How to test


[Link to Probo site](https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2308.probo.build/)

- Confirm that all news and event content type displays and views appear as expected (especially the updated Row display of events).
- Confirm that editing images in news and event nodes works as expected.
- Confirm that applying Card Image and Square Aspect Ratio cropping to thumbnail images works properly.
- Confirm that the new demo module content is adequate and appropriate.

Previous feedback requested:
  - Is it correct that the Grid View for events should be left as is? If so, is there a better way to provide a new Grid View with Images for events that does not require creating a new view mode for nodes?
    - Answer from 3-29 discussion: Yes, leave the Grid View appearance as is. We can consider using preprocessing to display the two views differently.
  - Should the Grid Sidebar view for events include the thumbnail images as well? Should an alternate sidebar view with images be created instead?
    - Answer from 3-29 discussion: No, leave the Grid Sidebar as is. We can add thumbnails to it later if desired.

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
